test/js/ez-field-edit.js

```diff
@@ -0,0 +1,113 @@
+describe('ez-field-edit', function() {
+    let element;
+    let elementWithMultipleInputs;
+    let elementWithoutLabel;
+
+    beforeEach(function () {
+        element = fixture('BasicTestFixture');
+        elementWithMultipleInputs = fixture('BasicTestFixtureMultipleInputs');
+        elementWithoutLabel = fixture('BasicTestFixtureInputWithoutLabel');
+    });
+
+    it('should be defined', function () {
+        assert.equal(
+            window.customElements.get('ez-field-edit'),
+            element.constructor
+        );
+    });
+
+    describe('properties', function () {
+        describe('`invalid`', function () {
+            it('should default to false', function () {
+                assert.notOk(element.invalid);
+            });
+
+            describe('set', function () {
+                it('should be reflected to an attribute', function () {
+                    element.invalid = true;
+
+                    assert.isTrue(
+                        element.hasAttribute('invalid')
+                    );
+                });
+            });
+        });
+    });
+
+    describe('validity', function () {
+        it('should get invalid', function () {
+            element.querySelector('input').value = 'invalid value';
+            element.dispatchEvent(new CustomEvent('input'));
+            assert.isTrue(
+                element.invalid
+            );
+        });
+
+        it('should get valid', function () {
+            element.invalid = true;
+            element.querySelectorAll('input').value = 42;
```

Comment:
this does not what you want to do here :) `querySelectorAll` returns a `NodeList` on which you add a `value` property with the value 42 ie you are not setting the input value :)